micro-rdk-nmea/src/parse_helpers/parsers.rs

```diff
@@ -11,40 +11,42 @@ use super::{
 /// Cursor that consumes can consume bytes from a data vector by bit size.
 pub struct DataCursor {
     data: Vec<u8>,
+    // the amount of bits by which the previously read field overflowed into
+    // the first byte
+    bit_offset: usize,
 }
 
 impl DataCursor {
     pub fn new(data: Vec<u8>) -> Self {
-        DataCursor { data }
+        DataCursor {
+            data,
+            bit_offset: 0,
+        }
     }
 
     pub fn read(&mut self, bit_size: usize) -> Result<Vec<u8>, NmeaParseError> {
-        if bit_size / 8 > self.data.len() {
+        let bits_to_read = bit_size + self.bit_offset;
+        if bits_to_read / 8 > self.data.len() {
```

Comment:
I think this won't work, because if you had one byte left, but `bits_to_read` was 15, this check would pass even though you only had 8 bits of data left.

Maybe `if bits_to_read > self.data.len() * 8` is clearer?